.github/workflows/qa.yaml

```diff
@@ -94,13 +94,13 @@ jobs:
         run: |
           set -eu
 
-          go test -coverpkg=./... -coverprofile=/tmp/coverage.out -covermode=set ./... -shuffle=on
+          ./gotestcov ./...
       - name: Run tests (with race detector)
         run: |
           go test -race ./...
       - name: Upload coverage to Codecov
         uses: codecov/codecov-action@v3
         with:
-          file: /tmp/coverage.combined
+          file: coverage/coverage.out.filtered
```

Comment:
I would avoid writing in the current project directory for accessing the coverage. I think in the CI in particular, redirecting this to /tmp is a better approach, which rely on gotestcov to support dedicated cover profile directory.